scanner/src/funTest/kotlin/experimental/PostgresNestedProvenanceStorageFunTest.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.experimental
+
+import com.opentable.db.postgres.embedded.EmbeddedPostgres
+
+import io.kotest.core.spec.Spec
+import io.kotest.core.test.TestCase
+
+import java.time.Duration
+
+private val PG_STARTUP_WAIT = Duration.ofSeconds(20)
+
+class PostgresNestedProvenanceStorageFunTest : AbstractNestedProvenanceStorageFunTest() {
+    private lateinit var postgres: EmbeddedPostgres
+
+    override fun beforeSpec(spec: Spec) {
+        postgres = EmbeddedPostgres.builder().setPGStartupWait(PG_STARTUP_WAIT).start()
```

Comment:
Just a suggestion: There is some duplication to setup the embedded test database, clean up the schema, etc. for all tests using an embedded Postgres DB. It could be beneficial to extract this logic into a test helper class. Due to the inheritance structures required by some tests, it will not be possible to have a common base class for these tests; but maybe something like [Kotest's listeners](https://github.com/kotest/kotest/blob/master/doc/listeners.md) could be used to provide the required setup code in a convenient way.

Reply:
Good idea, I have added a commit for that.